pkg/util/limiter/memory_tracker_middleware.go

```diff
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package limiter
+
+import (
+	"net/http"
+)
+
+// MemoryTrackerMiddleware adds an unlimited MemoryConsumptionTracker to the request context.
+type MemoryTrackerMiddleware struct{}
```

Comment:
[nit] I'd make it super clear that this will not enforce a memory consumption limit:

```suggestion
type UnlimitedMemoryTrackerMiddleware struct{}
```